### PR TITLE
resource_control: calibrate resource support sysbench

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

Issue Number: ref #38825

Problem Summary:

### Check List

Tests 

- [X] Unit test
- [X] Manual test (add detailed scripts or steps below)

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
calibrate resource support sysbench
```
image

Signed-off-by: husharp <jinhao.hu@pingcap.com>

### CODE REVIEW

**Comment by HuSharp** [comment]
@glorv @nolouch @JmPotato PTAL, thx!

File: executor/calibrate_resource.go [hunk 1]
```diff
@@ -21,40 +21,57 @@ import (
...
+	ast.OLTPREADWRITE: {
+		tidbCPU:       0.8,
+		kvCPU:         0.35,
+		readBytes:     units.MiB * 4.25,
+		writeBytes:    units.MiB / 3,
+		readReqCount:  1600,
+		writeReqCount: 1400,
+	},
+	ast.OLTPREADONLY: {
+		tidbCPU:       0.4,
```
[the remark this answers is not in the document]
Comment (HuSharp) [reply executor/calibrate_resource.go]: This is my fault, based on test data, updated to 2 here.

File: executor/calibrate_resource.go [hunk 2]
```diff
@@ -102,11 +115,9 @@ func (e *calibrateResourceExec) Next(ctx context.Context, req *chunk.Chunk) erro
 		return err
 	}
 
-	// we only support TPC-C currently, will support more in the future.
-	workload := defaultWorkload
-	baseCost, ok := workloadBaseRUCostMap[workload]
+	baseCost, ok := workloadBaseRUCostMap[e.workloadType]
 	if !ok {
-		return errors.Errorf("unknown workload '%s'", workload)
+		return errors.Errorf("unknown workload '%v'", e.workloadType)
```
[the remark this answers is not in the document]
Comment (HuSharp) [reply executor/calibrate_resource.go]: fixed as `%T`

File: parser/parser.y [hunk 3]
```diff
@@ -14816,8 +14827,32 @@ PlanReplayerStmt:
...
-		$$ = &ast.CalibrateResourceStmt{}
+		$$ = &ast.CalibrateResourceStmt{
+			Tp: $3.(ast.CalibrateResourceType),
+		}
+	}
+
+CalibrateResourceWorkloadOption:
+	/* empty */
+	{
+		$$ = ast.TPCC
```
[the remark this answers is not in the document]
Comment (HuSharp) [reply parser/parser.y]: Added a None type for marking

**Review by glorv — APPROVED** [review]
LGTM

**Review by JmPotato — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Pull request merged**